akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -646,9 +646,16 @@ private[persistence] trait Eventsourced
             case SelectedSnapshot(metadata, snapshot) =>
               val offer = SnapshotOffer(metadata, snapshot)
               if (recoveryBehavior.isDefinedAt(offer)) {
-                setLastSequenceNr(metadata.sequenceNr)
-                // Since we are recovering we can ignore the receive behavior from the stack
-                Eventsourced.super.aroundReceive(recoveryBehavior, offer)
+                try {
+                  setLastSequenceNr(metadata.sequenceNr)
+                  // Since we are recovering we can ignore the receive behavior from the stack
+                  Eventsourced.super.aroundReceive(recoveryBehavior, offer)
+                } catch {
+                  case NonFatal(t) =>
+                    try onRecoveryFailure(t, Some(offer))
```

Comment:
It would be much more useful to have the snapshot metadata logged. What about adding a case to handle that in `onRecoveryFailure`?

Reply:
I agree that would be more useful but due to backwards compatibility constraints we can't add a new method there. I think this is the best we can do, unless we want to add additional logging here that can't be overridden.

To be completely backwards compatible we can't even call `onRecoveryFailure` with the `SnapshotOffer` like this, because that parameter is documented to be an event and if application code relies on it only being event types it would break.
```
event the event that was processed in `receiveRecover`, if the exception was thrown there
```

Safe would be to use `None`, hopefully enough information is carried in the exception.

Reply:
I changed to `None` as the default `onRecoveryFailure` will log the persistence id and last sequence nr which allows to find out which exact snapshot caused the crash